ensenso_camera/src/virtual_object_handler.cpp

```diff
@@ -145,6 +145,10 @@ struct VirtualObjectMarkerPublisher
   {
     while (ensenso::ros::ok() && !stop)
     {
+      // Set to latest available timestamp
+      auto now = ensenso::ros::Time(0);
```

Comment:
Wouldn't the latest available timestamp be `ensenso::ros::now(nh)`? In our remaining code we use `Time(0)`  only to lookup the latest transform from a `tf` buffer, which is an implementation detail specific to the buffer. As far as I understand from this [ROS answer](https://answers.ros.org/question/231914/tf_echo-can-find-transform-but-waitfortransform-cant/), `Time(0)` is used for static transformations.

Reply:
To me it looks like there is some problem with the transformation from frame `ladle` to the `base_link` now. Could you maybe give me a short overview of the frames you are having in your setup and whether it is fixed or moving? 

Above you mentioned:
> When having moving object frame [...]

When the object itself is moving, its transformation should be frequently updated. If the object is moving, but e.g. always together with the robot hand, then I think the robot hand's transform should be frequently updated and the object transformation can be static.

Reply:
See attached image. `lift_ee` and `camera` are fixed.
![image](https://user-images.githubusercontent.com/26171571/215729208-3f4acd89-017e-4de3-b097-d4f4f1abcbbf.png)

Reply:
http://wiki.ros.org/rviz/DisplayTypes/Marker 
> Note that the timestamp attached to the marker message above is ros::Time(), which is time Zero (0). This is treated differently by RViz than any other time. If you use ros::Time::now() or any other non-zero value, rviz will only display the marker if that time is close enough to the current time, where "close enough" depends on TF. With time 0 however, the marker will be displayed regardless of the current time. 

It looks like you're right and you can use `Time(0)`. And if the objects moved in the virtual camera by doing so, even more reason to accept it ;)

It would be nice though, if we would initialize the timestamp with `Time(0)` once [here](https://github.com/ensenso/ros_driver/blob/master/ensenso_camera/src/virtual_object_handler.cpp#L57) and not update it later on in `run()`.

Reply:
Nice, makes more sense now :)
I can do the changes :+1: